src/Features/Core/Portable/Diagnostics/Analyzers/NamingStyles/WellKnownSymbolSpecifications.cs

```diff
@@ -0,0 +1,184 @@
+                ModifierKindEnum.IsStatic, Private.Concat(Internal).ToList(), "{AC995BE4-88DE-4771-9DCC-A456A7C02D89}"));
+
+
+            return result;
+        }
+
+        public static List<NamingStyle> GetWellKnownNamingStyles()
+        {
+            var result = new List<NamingStyle>();
+
+            result.Add(new NamingStyle()
+            {
+                ID = Guid.Parse("{F408FCBC-0C17-4974-BE20-BCB62BC81A62}"),
+                Name = FeaturesResources.Camel_Case,
+                CapitalizationScheme = Capitalization.CamelCase
+            });
+
+            result.Add(new NamingStyle()
+            {
+                ID = Guid.Parse("{87E7C501-9948-4B53-B1EB-A6CBE918FEEE}"),
+                Name = FeaturesResources.Camel_Case,
+                CapitalizationScheme = Capitalization.PascalCase
+            });
+
+            result.Add(new NamingStyle()
+            {
+                ID = Guid.Parse("{868DE8A1-CA8D-4190-B7E1-88CB38210C82}"),
+                Name = FeaturesResources.Pascal_Case,
+                CapitalizationScheme = Capitalization.FirstUpper
+            });
+
+            result.Add(new NamingStyle()
+            {
+                ID = Guid.Parse("{C5032A64-92E6-4A3B-9152-F01272DF550D}"),
+                Name = FeaturesResources.All_uppercase,
+                CapitalizationScheme = Capitalization.AllUpper
+            });
+
+            result.Add(new NamingStyle()
+            {
+                ID = Guid.Parse("{AFF92434-B78F-413C-A6EF-ED5007F45167}"),
+                Name = FeaturesResources.All_lowercase,
+                CapitalizationScheme = Capitalization.AllLower
+            });
+
+            result.Add(new NamingStyle()
+            {
+                ID = Guid.Parse("{308152F2-A334-48B3-8BEC-DDEE40785FEB}"),
+                Name = string.Format(FeaturesResources.Ends_with_0, "Async"),
+                Suffix = "Async"
+            });
+
+            result.Add(new NamingStyle()
+            {
+                ID = Guid.Parse("{54B14419-268B-4CC2-975F-526CC8460298}"),
+                Name = string.Format(FeaturesResources.Begins_with_0, "_"),
+                Prefix = "_"
+            });
+
+            result.Add(new NamingStyle()
+            {
+                ID = Guid.Parse("{1CADEA25-F609-449A-A7EF-B9080BCA2163}"),
+                Name = string.Format(FeaturesResources.Begins_with_0, "s_"),
+                Prefix = "s_"
+            });
+
+            result.Add(new NamingStyle()
+            {
+                ID = Guid.Parse("{2ACD535E-4C74-4838-9A0A-02B337DADFAA}"),
+                Name = string.Format(FeaturesResources.Begins_with_0_camel_case, "_"),
+                Prefix = "_"
+            });
+
+            result.Add(new NamingStyle()
+            {
+                ID = Guid.Parse("{0376DC66-18E6-40FF-8266-A54F5E1D15FC}"),
+                Name = string.Format(FeaturesResources.Begins_with_0_camel_case, "s_"),
```

Comment:
I'll bring back up the `t_` thing from the dotnet team. I really want to be able to support them in that without building that naming style as one of the default options (seems overkill, unless we really want to be recommending that pattern).